graylog2-web-interface/src/views/components/SearchBar.tsx

```diff
@@ -95,7 +96,11 @@ const defaultOnSubmit = ({ timerange, streams, queryString }, currentQuery: Quer
     .query(createElasticsearchQueryString(queryString))
     .build();
 
-  return QueriesActions.update(newQuery.id, newQuery);
+  if (!currentQuery.equals(newQuery)) {
+    return QueriesActions.update(newQuery.id, newQuery);
+  }
+
+  return SearchActions.refresh();
```

Comment:
We need to return a promise here, otherwise formik will not update `isSubmitting`. Since we expect the action to return a promise, I created a new method for the store, to actually return a promise.